### PR TITLE
[#2981] Ported over Raspbian support.

### PR DESCRIPTION
**Pull request by dumol**
## Problem

There was a need for `paver.sh` changes in `python-packages` for supporting Raspbian.
## Solution

Port the changes from `python-packages` for supporting Raspbian.
## How to test

Please review changes.
Run the tests.

reviewer: @adiroiban

### CODE REVIEW

**Comment by adiroiban** [comment]
This is ok.

Please note that for Python distribution I would like to have a generic detect_os method which will return the actual OS / ARCH of the system.

On top of that, we can have another layer which can define how different OSes are mapped to generic builds.

For now, I don't know how this can  or should be done so is ok to have this merge.

But, to have this merged we should check that the new code works and can be used to produce a new python distribution.

I have launched a manual build http://build.chevah.com/builders/python-distribution-raspbian-7/builds/0 which I guess it will take a while.

I have updated the documentation with info about triggering a build.

Please check that it make sense.

Do we also need to migrate the test? as the test is failing

http://build.chevah.com/builders/python-distribution-raspbian-7/builds/0/steps/test/logs/stdio

needs-changes

**Comment by dumol** [comment]
I think the OS / ARCH detection and special treatment for select Linux distros is to be discussed in https://trac.chevah.com:10443/ticket/2935.

The test fails because I'm not done with this, I still have to port the changes to the tests. Thank you for updating the documentation for triggering a build. I think there should be more documentation, as the build system is not clear even for me, and I'm familiar with `python-package`. People expect things to be straightforward.

I'm also wondering how to increment the version for this. Would this be `PYTHON_VERSION='2.7.8+chevah2'`? 

What are these though?

```
PAVER_VERSION='1.2.1'
# It requires a brink version newer than 0.56
BRINK_VERSION='0.56.0
```

And why is the brink version at 0.56.0 when brink master is still at 0.55.x?

Thanks.

**Comment by dumol** [comment]
Not sure how to continue with this… There are three different errors when testing this through buildbot, but none of them are related to the changes in this branch. Namely:
1. `python-distribution` doesn't install needed packages, as python-package does in RHEL, SLES, Ubuntu LTS.
2. Non-linux builds fail because `libffi` build part is broken.
3. Older `wget` versions (on RHEL5 and SLES 11) fail with:

```
--2015-09-18 14:37:07--  https://bootstrap.pypa.io/get-pip.py
Resolving bootstrap.pypa.io... 23.235.43.175
Connecting to bootstrap.pypa.io|23.235.43.175|:443... connected.
ERROR: certificate common name `*.c.ssl.fastly.net' doesn't match requested host name `bootstrap.pypa.io'.
To connect to bootstrap.pypa.io insecurely, use `--no-check-certificate'.
Unable to establish SSL connection
```

For problem 1, I only had to install packages manually on RHEL6, the others had them installed already. I think this is because the automatically-installed packages are not uninstalled when a build of `python-package` fails.

For problem 2, I could reconfigure buildmaster to only test on Linux bots for now, perhaps defining a _review_ group for _python-distribution_?

For problem 3, I could install a newer `wget` in `/usr/local` on RHEL5 and SLES11. Would that be fine?

**needs-review**

**Comment by adiroiban** [comment]
To  continue working on this we first need to address the blocking issues :(
1. Create a ticket to fix python-distribution so that it will install the required packages.

---
1. Update the review group to only run the systems supported by python-distribution

---
1. For 3, we can have a newer wget or run it with --no-check-certificate

Do we still need RHEL5 / Ubuntu and SLES 11 ?

Maybe they work fine with the generic linux-build.
In this case we will have a single linux-x64 python-distribution build.

We will still continue to have RHEL5 / SLES11 tests for the server project.

---

I see that http://build.chevah.com/builders/python-distribution-raspbian-7 is green, so we can just merge this code and work on the remaining issues without blocking it.

Thanks!

changes-approved

File: test/test_python_binary_dist.py [hunk 1]
```diff
@@ -72,6 +72,13 @@ def get_allowed_deps():
                 'libtinfo.so.5',
                 ])
         else:
+            # Raspbian is a special case... If this file exists, we assume the
+            # distro passes the more stringent checks during the build phase.
```
Comment (adiroiban) [inline test/test_python_binary_dist.py]: just a small note.
This comment is actually related to the code inside the `if os.path.isfile("/etc/rpi-issue"):` block so it should be indented at the same level as the `allowed_deps.extend(['libcofi_rpi.so',` code.

Comment (dumol) [inline test/test_python_binary_dist.py]: Good catch, thanks!

**Comment by dumol** [comment]
1. If we intend to move to generic Linux builds, I think we should also ditch installing/uninstalling packages.  First of all, I think it's a bad idea to feed spoon the user and install packages on his behalf. And most importantly, it's broken as it is in `python-package`, because packages are not uninstalled if the build fails. Which inevitably happens sooner or later, so now all our build slaves have the full stack required for `python-package` installed. Checking for deps is a better strategy, /me thinks, and there is a ticket for improving that.
2. I have reconfigured the groups for `python-distribution` for our build master.
3. I have compiled and installed the latest wget in `/usr/local/` on RHEL 5 and SLES 11.

We can't use generic Linux builds for RHEL5 and SLES 11 because they only have OpenSSL 0.9.x.

**Comment by adiroiban** [comment]
1. True. We still need some helper code for our buildslave to make sure they have the right environment for running the build. I am ok with checking for deps.
2. Thanks.
3. Thanks.

I think that we can have raspbian support enabled and have this merged.

Thanks!

**Pull request closed without merging**